example/float128_snips.cpp

```diff
@@ -3,40 +3,91 @@
 //  LICENSE_1_0.txt or copy at https://www.boost.org/LICENSE_1_0.txt
 
+// Demonstrations of using Boost.Multiprecision float128 type.
+// Contains Quickbook markup in comments.
+
 //[float128_eg
 #include <boost/multiprecision/float128.hpp>
 #include <boost/math/special_functions/gamma.hpp>
 #include <iostream>
 
 int main()
 {
-   using namespace boost::multiprecision;
+   using namespace boost::multiprecision; // Potential to cause name collisions?
+   // using boost::multiprecision::float128; // is safer.
 
-   // Operations at 128-bit precision and full numeric_limits support:
+/*`The type float128 provides operations at 128-bit precision with
+[@https://en.wikipedia.org/wiki/Quadruple-precision_floating-point_format#IEEE_754_quadruple-precision_binary_floating-point_format:_binary128 Quadruple-precision floating-point format]
+and have full `std::numeric_limits` support:
+*/
    float128 b = 2;
-   // There are 113-bits of precision:
+//` There are  15 bits of (biased) binary exponent and 113-bits of significand precision
    std::cout << std::numeric_limits<float128>::digits << std::endl;
-   // Or 34 decimal places:
+//` or 33 decimal places:
    std::cout << std::numeric_limits<float128>::digits10 << std::endl;
-   // We can use any C++ std lib function, lets print all the digits as well:
-   std::cout << std::setprecision(std::numeric_limits<float128>::max_digits10) 
-      << log(b) << std::endl; // print log(2) = 0.693147180559945309417232121458176575
-   // We can also use any function from Boost.Math:
+   //` We can use any C++ std library function, so let's show all the at-most 36 potentially significant digits, and any trailing zeros, as well:
+    std::cout.setf(std::ios_base::showpoint); // Include any trailing zeros.
+   std::cout << std::setprecision(std::numeric_limits<float128>::max_digits10)
+      << log(b) << std::endl; // Shows log(2) = 0.693147180559945309417232121458176575
+   //` We can also use any function from Boost.Math, for example, the 'true gamma' function `tgamma`:
    std::cout << boost::math::tgamma(b) << std::endl;
-   // And since we have an extended exponent range we can generate some really large 
-   // numbers here (4.02387260077093773543702433923004111e+2564):
+   /*` And since we have an extended exponent range, we can generate some really large
+    numbers here (4.02387260077093773543702433923004111e+2564):
+   */
    std::cout << boost::math::tgamma(float128(1000)) << std::endl;
-   //
-   // We can declare constants using GCC or Intel's native types, and the Q suffix,
-   // these can be declared constexpr if required:
+   /*` We can declare constants using GCC or Intel's native types, and literals with the Q suffix, and these can be declared `constexpr` if required:
+   */
    /*<-*/
 #ifndef BOOST_NO_CXX11_CONSTEXPR
    /*->*/
-   constexpr float128 pi = 3.1415926535897932384626433832795028841971693993751058Q;
+   constexpr float128 pi = 3.14159265358979323846264338327950Q;
```

Comment:
That doesn't seem to have enough digits anymore?

Reply:
I reduced this to 36 decimal digits to match the number with the implied precision of float128 of std::numeric_limits::<float128>max_digits10 = 36 as I felt this was clearer.  But I could revert it if you don't think that this helps.

Reply:
There aren't 36 digits?

Reply:
You are, as always, this is digits10 rather than max_digits10.  I'll change it.